superset/explore/permalink/commands/create.py

```diff
@@ -39,11 +39,16 @@ def __init__(self, actor: User, state: Dict[str, Any]):
     def run(self) -> str:
         self.validate()
         try:
-            dataset_id = int(self.datasource.split("__")[0])
-            check_access(dataset_id, self.chart_id, self.actor)
+            datasource = self.datasource.split("__")
+            datasource_id: int = int(datasource[0])
+            datasource_type: str = datasource[1]
```

Comment:
I really wanted to do a nice tuple unpacking here, but unfortunately mypy wouldn't have it.